### ISSUE TITLE
Restore the Cut, Copy and Paste buttons as a new (hidden by default) toolbar

### ISSUE THREAD

**Issue by petersampsonaudacity** [audacity/audacity#3693]
### Your idea

Restore the Cut, Copy and Paste buttons as a new toolbar that would be 

a) hidden by default to keep the revised 3.2 GUI clean

b) available for users to make visible in the upper tooldock with the View>Toolbars command

**Oversize mockup**
![image](https://user-images.githubusercontent.com/50205984/192318204-a46bcc69-6697-4e3d-9966-3964f6386dc6.png)

**In situ mockup**
![image](https://user-images.githubusercontent.com/50205984/192319720-87b5fbca-b7cd-4995-b961-5fee22129631.png)

Some users are already complaining about the loss of these buttons on the Forum:
https://forum.audacityteam.org/viewtopic.php?f=46&t=127074&p=456464
and this is causing some of them to revert to older versions of Audacity just to get the buttons back _(which is not ideal)_

### Problem to be solved

1)  some users are complaining about the removal of the Cut, Copy and Past buttons ffrom the Edit toolbar

2) some of those users are thus choosing to revert to older versions of Audacity, just to get the buttons back

### Prior art

Microsoft apps that I use have such buttons _(in addition to the "standard" shortcuts)_ - e.g.:
1. Microsoft Office: Word and Excel
2. Microsoft Paint

### Additional context

See Forum threads regarding the loss of these buttons:
**(Audacity 3.2.0) Please Bring Back Cut, Copy, & Paste Buttons]**
https://forum.audacityteam.org/viewtopic.php?f=70&t=127081#p456403)#

And:
**audacity old version download**
https://forum.audacityteam.org/viewtopic.php?f=46&t=127074

**Comment by petersampsonaudacity** [comment]
Flagging @Tantacrul as a UX issue

**Comment by Tantacrul** [comment]
Oh dear, I've never been called out as a UX issue before 😭

**Comment by petersampsonaudacity** [comment]
> Oh dear, I've never been called out as a UX issue before 😭

No - you're a UX champion who can help these button-pushers get their buttons back - they'd love you for it   ;-)

**Comment by Tantacrul** [comment]
No matter the function, if you remove something, someone's workflow will be interrupted. However, in this case there are already three other ways one can cut, copy or paste. It's just not something we should burden the interface with.

**Comment by petersampsonaudacity** [comment]
A comment from a user on the Forum 

@LWinterberg Leo wrote in this thread
[quote]buttons are primarily used by people who don't use keyboard shortcuts, so by and large fairly novice users. [/quote]

User scott967 responded
[quote]Curious as to the basis for this assertion?  I mean, I do UI work for a different open source project, and have never encountered this attitude, except maybe by devs.[/quote]

**Comment by petersampsonaudacity** [comment]
> Cut/copy/paste buttons are primarily used by people who don't use keyboard shortcuts, so by and large fairly novice users.

I do note that Microsoft in their Office products still retain cut/copy/paste buttons _(and in a very prominent position)_ - and I'm pretty sure that not all of their users are ""fairly novice users"

Excel for Example
![image](https://user-images.githubusercontent.com/50205984/194052678-8cb63379-02d5-4742-bd02-cfe5c2d71366.png)


And Word
![image](https://user-images.githubusercontent.com/50205984/194052964-1e2e4fb1-6a4c-407d-8bc2-e262ce863f7e.png)

**Comment by Tantacrul** [comment]
@LWinterberg / @DilsonsPickles: Let's create a new panel specifically for these options, which will be hidden by default.

Then we can incorporate them into our show/hide widget for the Qt interface of Audacity too. Cheers!

**Comment by petersampsonaudacity** [comment]
I note that Paint 3D has a four-button edit toolbar with cut, copy, paste and delete - so maybe a Delete button could be added to this new (hidden by default) toolbar:
![image](https://user-images.githubusercontent.com/50205984/198620569-00ba0ee0-5003-438b-b99f-1667a94a413d.png)

**Comment by petersampsonaudacity** [comment]
> so maybe a Delete button could be added to this new _(hidden by default)_ toolbar

Double-size mockup
![image](https://user-images.githubusercontent.com/50205984/198836722-4fe87d80-e218-4160-9b19-79c0f23b269e.png)

**Comment by DilsonsPickles** [comment]
Hi gang, not sure what happened to the previous image I posted...

Here's the Figma link to the design specification: https://www.figma.com/file/uYdHVW7Hhkha51ZGGfC3RF/09---Tools?node-id=328%3A4323

Or [click here](https://www.figma.com/file/uYdHVW7Hhkha51ZGGfC3RF/09---Tools?node-id=328%3A4323)


**Note:** This is our first run of sharing Figma projects outside of our team account so please forgive any teething issues. 

@petersampsonaudacity I will be adding you to the file via your email, so if the links don't work then just keep an eye on your inbox. Please let me know if you run into any trouble. 

There is a comment system in Figma but I fear we'll lose track of feedback, so please continue to add your feedback to the ticket!

**Comment by petersampsonaudacity** [comment]
@DilsonsPickles  looks good to me as that double-height toolbar - and thanks for adding the delete button  :-))

_I note that it does not require additional top tooldock space (as per current 3.2.2/3.3.3 default) as you appear have dropped the similar sized Tools toolbar - which I understand from @Tantacrul is due to go soon anyway...

But I also note that you still have in your second design image an entry in the View Toolbars menu for Tools Toolbar  -  I assume this is just an oversight._

**Comment by DilsonsPickles** [comment]
@petersampsonaudacity great spot! I'll remove the Tools Toolbar menu item now. Thanks.

**Comment by Ryan-Miller-Audacity** [comment]
Tested on both MacOS and Win10 via build: https://github.com/audacity/audacity/actions/runs/3757090295. 

Cut, Copy, Paste toolbar is properly hidden by default. 
Toolbar properly displays when selected to view
Selection is remembered between startups
Buttons function as expected and functionality matches keyboard shortcuts

One note: The previous discussion on this item seems there means to be a delete button included with the change; however, that is currently not the case as there is no delete button. Was this left out intentionally @ksoze95 ? Or will it be added later and need a separate issue for tracking?

**Comment by crsib** [comment]
I've asked on Monday and was told that there is no need for Delete button

**Comment by Ryan-Miller-Audacity** [comment]
Gotcha, moving along then.

**Comment by petersampsonaudacity** [comment]
No, sorry, not moving along then ...

When I started in this ENH thread it was in response to many requests from users on the Forum - and you will note from the initial post in this thread it was initially just a three button Cut/Copy/Paste - AND these were in a single height kinear toolbar/

Further discussion on the Forum resulted in users also asking to add a Delete button - which I added to this thread at the time still retaining a single-height linear toolbar.

@DilsonsPickles was subsequently assigned by @Tantacrul  to design the new toolbar - and in his Figma design, which he share with me, he changed it to the double-height four-square design with the four buttons Cut/Copy/Paste/Delete - which looked good to me.

### Functionality
>I've asked on Monday and was told that there is no need for Delete button
I really don't buy that argument as Delete is not the same as Cut -it is similar but fundamentally different in that Cut replaces the existing contents of the clipboard (which the user may not want), Delete does _**not**_ do that.

### Aesthetics
Apart from the functionality argument, there is also the aesthetic one.  It just looks plain daft and incomplete with an empty useless square in the four square design.

If you are insistent on having just the three buttons then it would be aesthetically much better to return to the single heiight linear design:

**Scaled x2**
![image](https://user-images.githubusercontent.com/50205984/209328930-bf7f1ac7-94ed-410d-9dc0-f95854d8f749.png)

**Comment by AnitaBats** [comment]
@Tantacrul please take a look on the comment from @petersampsonaudacity and let us know if any changes are needed
Thanks!

**Comment by Tantacrul** [comment]
@AnitaBats - looking now...

**Comment by Tantacrul** [comment]
@petersampsonaudacity:

Yes, I think we _should_ have a delete button but since we have established Tuesday next week as our release date - we do not want to include the fix now. Instead, we'll add the delete button in our 3.3 release instead, which will be in late March.

Hope this is agreeable to you, @petersampsonaudacity!

**Comment by petersampsonaudacity** [comment]
> Hope this is agreeable to you, @petersampsonaudacity!

More than agreeable - many thanks @Tantacrul    :-))

**Comment by petersampsonaudacity** [comment]
Testing on W10 with the latest 3.2.3 beta:  audacity-win-3.2.3-beta-20221223+ae85284-x64-msvc2022

### 1) Upgrading from 3.2.2
When I upgrade from 3.2.2 to 3.2.3
a) the Cut/Copy/Paste toolbar is not _**OFF**_ by default as it should be
b) it is placed on an additional double-height line in the upper tooldock
![image](https://user-images.githubusercontent.com/50205984/209437147-9afc220e-e0df-4d75-a26f-eec5e4953014.png)

**Comment by petersampsonaudacity** [comment]
Also note that the Paste button is on by default.

I think this is because I have stuff (text) on my windows clipboard - but there is nothing on my Audacity clipboard.

![image](https://user-images.githubusercontent.com/50205984/209437847-bc3d8748-8bd6-46b8-b2ee-a7b8bad87b7a.png)

**Comment by Tantacrul** [comment]
> ### 1) Upgrading from 3.2.2
> When I upgrade from 3.2.2 to 3.2.3 a) the Cut/Copy/Paste toolbar is not _**OFF**_ by default as it should be b) it is placed, all alone, on an additional double-height line in the upper tooldock with a lot of wasted space to the right ![image](https://user-images.githubusercontent.com/50205984/209437147-9afc220e-e0df-4d75-a26f-eec5e4953014.png)


It's quite important for the Cut/Copy/Paste toolbar to **not** be displayed in this circumstance. Thanks for finding and flagging this @petersampsonaudacity !

**Comment by crsib** [comment]
This is really weird, I will recheck Monday

**Comment by Tantacrul** [comment]
Confirmed about your first point, Peter. I installed this build (universal MacOS)
https://github.com/audacity/audacity/actions/runs/3767368881/jobs/6404828886

and got this:
<img width="1048" alt="image" src="https://user-images.githubusercontent.com/21022311/209446140-6e60179e-f77e-46c5-83df-6ccc942be9e5.png">

**Comment by Tantacrul** [comment]
One other point: has anyone noticed that the 'Share audio' button has no padding? @ksoze95 - any ideas?

**Comment by crsib** [comment]
I have fixed that couple times already, it keeps getting back

**Comment by Tantacrul** [comment]
Sorry Dmitry!

I'll leave you alone given it's Christmas Eve. 😀

**Comment by petersampsonaudacity** [comment]
> Question: how do I test **upgrading** from 3.2.2 to 3.2.3? Do I simply install 3.2.3 and make sure 3.2.2 in on my machine already first?

What I do and (what I have always  done for many years of Audacity QA testing) is
1. clear the audacity settings folder - delete the entire contents
2. Launch say 3.2.2
3. this sets up the config files in a 3.2.2 state
4. Exit 3.2.2
5. launch 3.2.3

Basically Audacity's behaviors are defined by the state of the audacity settings folder particularly the .cfg files therein.  This means that both versions can be installed on your computer before this test is done, Step-1 is the critical one,

Of course if the user after step 5 actually installs 3.2.3 on Windows with the EXE installer and chooses the option to reset config settings then thy will effectively get a clean "virgin" 3.2.3 launch.  Don't forget that the Mac and the Linux (and indeed the Windows zip) installations do not offer the config reset - only the Windows EXE installer does that _(and don't ask me why...)_

**Comment by DavidBailes** [comment]
> I have fixed that couple times already, it keeps getting back

In ToolManger::ReadConfig(), just add a case for setting bShownByDefault to be false for ndx == CutCopyPasteID?

      bool bShownByDefault = true;
      int defaultDock = TopDockID;

      if( ndx == SelectionBarID )
         defaultDock = BotDockID;
      if( ndx == MeterBarID )
         bShownByDefault = false;
      if( ndx == ScrubbingBarID )
         bShownByDefault = false;
      if( ndx == DeviceBarID )
         bShownByDefault = false;
      if( ndx == TimeBarID )
         defaultDock = BotDockID;

**Comment by crsib** [comment]
That was about Share Audio toolbar witdth

> In ToolManger::ReadConfig()

Not as simple I think, but quite similar

**Comment by DavidBailes** [comment]
> That was about Share Audio toolbar witdth

I'm easily confused!

**Comment by crsib** [comment]
> I'm easily confused!

Nah, our bad :-) The topic ... went offtopic :-)

BTW it appears that your proposed solution is sufficient, thank you a lot!

**Comment by petersampsonaudacity** [comment]
Testing with @crsib Dmitry's latst branch build: audacity-win-3.2.3-alpha-20221226+684a907-x64

1) For a virgin install of 3.2.3, the cut/copy/paste toolbar is hidden by default as expected

2) For an upgrade from 3.2.2 to3.2.3, the cut/copy/paste toolbar is now properly hidden by default as expected 

3) But if the cut/copy/paste toolbar is then turned on with View>Toolbars - then the meter toolbars are pushed down onto a new single-height line in the upper tooldock - which is not the ddefault 3.2.3 laout where all toolbars fit in a single double-height line in the upper tooldock.
![image](https://user-images.githubusercontent.com/50205984/209542593-e103dc70-9cd2-4220-8790-97c067eeb22e.png)

**Comment by crsib** [comment]
This is kind of expected. 

> default the meters have been shortened 

Actually, the default window width was extended

**Comment by petersampsonaudacity** [comment]
ASIDE### 
The use-case b) above is not strictly accurate as;
c) the positioning is retained
d) the meters are returned to default length - _this is an already logged bug_

**Comment by petersampsonaudacity** [comment]
### Further aside
Personally I like my meter toolbars on such a single-height line below the double-height one as I like to stretch out my recording meter for production usage:
![image](https://user-images.githubusercontent.com/50205984/209544407-18e2dbe9-7f12-488f-ba6f-c363ca9b99ea.png)

**Comment by petersampsonaudacity** [comment]
Testing on W10 with latest 3.2.3 Beta:  audacity-win-3.2.3-beta-20221226+387f691-x64-msvc2022

All looks good now both fir virgin install of 3.2.3 and for the upgrade from 3.2.2.

_Oh, and the **Share Audio** button is now resized to get its padding back._

**Comment by crsib** [comment]
Thank you, Peter!

**Comment by petersampsonaudacity** [comment]
@Tantacrul @DilsonsPickles 

Given that for 3.2.3 the upper tooldock and the window size has been changed to accommodate sufficient space in the single double-height upper tooldock, I find myself wondering why its has to be **off** by default
![image](https://user-images.githubusercontent.com/50205984/209659181-0adcd4ec-a6eb-4176-ab33-9fa6a5eb0b9e.png)

Surely the default upper tooldock would look better filled with it turned **on** by default instead, with only minimal padding on the right?
![image](https://user-images.githubusercontent.com/50205984/209659716-2db7c07a-7537-4928-92c0-4aabd7145ad0.png)

**Comment by Tantacrul** [comment]
I saw some message from @petersampsonaudacity asking 'why does this toolbar need to be off by default?'

Answer: because it is redundant to the vast majority of people.

**Comment by petersampsonaudacity** [comment]
> I saw some message from @petersampsonaudacity asking 'why does this toolbar need to be off by default?'

But then @Tantacrul I quickly withdrew that musing ...   ;-))

**Comment by petersampsonaudacity** [comment]
@Tantacrul @crsib @LWinterberg 

I have spent this afternoon updating the alpha 3.2.3 Manual with the bew Cut/Copy/Paste toolbar.

This is the page: https://alphamanual.audacityteam.org/man/Cut/Copy/Paste_Toolbar

Linked to from here:  
a) https://alphamanual.audacityteam.org/man/View_Menu
b) https://alphamanual.audacityteam.org/man/Toolbars_Overview
c) https://alphamanual.audacityteam.org/man/View_Menu:_Toolbars

This also led to a tidying up and updating of those pages that link to the new page.

Are we going to pill the alpha manual for the 3.2.3 release _(something we neglected to do for 3.2.2)_ ?

**Comment by crsib** [comment]
I have updated the manual for 3.2.2. Wrong version number is some kind of a bug I think

**Comment by petersampsonaudacity** [comment]
@crsib  Dmitry

> I have updated the manual for 3.2.2. 

But I changed the layout of the landing page for 3.2,2 making it a shorter reference page with subpages for
a) how-tos
b) tutorials

>Wrong version number is some kind of a bug I think

That was a bug initially in 3.2.1 (my mistake I bumped the alpha manual to 3.3)  @LWinterberg  corrected that in the alpha manual long before 3.2.2 was released

Neither of those changes can be seen in the online manual for 3.2.2

But when I access the online Manual from the latest 3.2.3 Beta I do see the properly updated, editable,  Manual _(including the changes I made today for the Cut/Copy/Paste toolbar)_

**Comment by crsib** [comment]
You can observe the new page, for example, here https://github.com/audacity/audacity-manual/blob/master/manual/man/cut/copy/paste_toolbar.html

**Comment by petersampsonaudacity** [comment]
@crsib   I am totally trusting your Magicke Spelles Dmitry  ;-))
